### PR TITLE
unique refresh token ID for token inline hook

### PR DESCRIPTION
**Pull request by susanharper-okta**
## Description:
- **What's changed?** adding info around unique refresh token ID
- **Is this PR related to a Monolith release?** Yes 2023.05.0

### Resolves:

* [OKTA-578302](https://oktainc.atlassian.net/browse/OKTA-578302)

### CODE REVIEW

**Comment by acrolinxdevdocs-okta** [comment]
## Acrolinx score **DRAFT 06.01.22**
 
A minimum Acrolinx Score of **40** is required. The total score is an average of the subscores.

Select **Total score** to review the Acrolinx scorecard for your article. Try to increase your individual scores, ex. Correctness. Your article will be clearer and more consistent with Microsoft standards.
 
| Article | **Total score**<br />**Required:70** | Word and phrases<br />(Brand, terms)<br />Preferred: 40 | Correctness<br />(Spelling, grammar)<br />Preferred: 40 | Clarity<br />(Readability)<br />Preferred: 10 | Inclusive language<br />(+ accesibility)<br />Preferred: 10 |
|---------|:------------------------------------:|:--------------------------------------------------------:|:--------------------------------------------------------:|:---------------------------------------------:|:------------------------------------------------------------:|
| [:white_check_mark:](# "Successfully processed") [packages/@okta/vuepress-site/docs/reference/token-hook/index.md](https://github.com/okta/okta-developer-docs/blob/cada1f65a1dd8fef46367c7c9318271e0073fda6/packages%2F%40okta%2Fvuepress-site%2Fdocs%2Freference%2Ftoken-hook%2Findex.md) | [68](https://okta.acrolinx.cloud/api/v1/checking/scorecards/8d5e17f1-c951-4587-93cb-7d250c9df876) | 80  | 56  | 33  | 96  |

  - The results from this response were procced by the [hugo-targets1.solution.acrolinx.sh](https://hugo-targets1.solution.acrolinx.sh) Acrolinx Cloud Host.
  - Scorecard links by default are only good for 7 days. [Link for this commit's Content Analysis Dashboard](https://okta.acrolinx.cloud/api/batch/gen.github.e44900b1-0907-42a5-b256-74181dc97026?csBaseUrl=https%3A%2F%2Fokta.acrolinx.cloud&X-Acrolinx-Client-Locale=en)

File: packages/@okta/vuepress-site/docs/reference/token-hook/index.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ This type of inline hook is triggered when OAuth 2.0 and OpenID Connect (OIDC) t
 
 This functionality can be used to add data that is sensitive, calculated at runtime, or complexly-structured and not appropriate for storing in Okta user profiles. Data added this way is never logged or stored by Okta. As an example, tokens minted for a medical app could be augmented with confidential patient data provided by your external service and not stored in Okta.
 
-In addition to adding custom claims, you can modify or remove an existing custom claim or an OIDC standard profile claim. You can also update how long an access token or an ID token is valid.
+In addition to adding custom claims, you can modify or remove an existing custom claim or an OIDC standard profile claim. You can update how long an access token or an ID token is valid. You can also refresh the access token using a unique refresh token ID sent in the request. That ID is stored and when the refresh token request is sent, the unique refresh token ID is matched against what is stored before issuing the refreshed access token.
```
Comment (anweshdatla-okta) [inline packages/@okta/vuepress-site/docs/reference/token-hook/index.md:26]: you can remove this doc change here.

Comment (anweshdatla-okta) [reply packages/@okta/vuepress-site/docs/reference/token-hook/index.md:26]: next to ###scopes you can add 
### data.refresh_token
Provides the refresh token ID, you could store the ID on the first access token request. When doing refresh token grant, its Id will match what was stored which should allow you to connect the two requests.

| Property | Description                   | Data Type                    |
|----------|-------------------------------|------------------------------|
| jti   | Refresh token ID | String     |

**Review by anweshdatla-okta — CHANGES_REQUESTED** [review]

File: packages/@okta/vuepress-site/docs/reference/token-hook/index.md (hunk 1 above)
Comment (anweshdatla-okta) [reply packages/@okta/vuepress-site/docs/reference/token-hook/index.md:26]: @venkatviswanathan-okta please check if this text needs correction.

File: packages/@okta/vuepress-site/docs/reference/token-hook/index.md [hunk 2]
```diff
@@ -327,7 +327,10 @@ Returning an error object causes Okta to return an OAuth 2.0 error to the reques
           "action": "GRANT"
         }
       }
-    }
+    },
+     "refresh_token": {
+         "jti": "oarob4a0tckCkGcyo1d6"
+      }
```
Comment (anweshdatla-okta) [inline packages/@okta/vuepress-site/docs/reference/token-hook/index.md:333]:
```suggestion
    },
    "refresh_token": {
       "jti": "oarob4a0tckCkGcyo1d6"
    }
```

**Comment by acrolinxdevdocs-okta** [comment]
## Acrolinx score **DRAFT 06.01.22**
 
A minimum Acrolinx Score of **40** is required. The total score is an average of the subscores.

Select **Total score** to review the Acrolinx scorecard for your article. Try to increase your individual scores, ex. Correctness. Your article will be clearer and more consistent with Microsoft standards.
 
| Article | **Total score**<br />**Required:70** | Word and phrases<br />(Brand, terms)<br />Preferred: 40 | Correctness<br />(Spelling, grammar)<br />Preferred: 40 | Clarity<br />(Readability)<br />Preferred: 10 | Inclusive language<br />(+ accesibility)<br />Preferred: 10 |
|---------|:------------------------------------:|:--------------------------------------------------------:|:--------------------------------------------------------:|:---------------------------------------------:|:------------------------------------------------------------:|
| [:white_check_mark:](# "Successfully processed") [packages/@okta/vuepress-site/docs/reference/token-hook/index.md](https://github.com/okta/okta-developer-docs/blob/7f150f5d4d33bc3756c0c3c3500b2348d069a433/packages%2F%40okta%2Fvuepress-site%2Fdocs%2Freference%2Ftoken-hook%2Findex.md) | [68](https://okta.acrolinx.cloud/api/v1/checking/scorecards/6853208a-7a9f-44d6-b282-d1b89f37a06d) | 80  | 57  | 34  | 96  |

  - The results from this response were procced by the [hugo-targets1.solution.acrolinx.sh](https://hugo-targets1.solution.acrolinx.sh) Acrolinx Cloud Host.
  - Scorecard links by default are only good for 7 days. [Link for this commit's Content Analysis Dashboard](https://okta.acrolinx.cloud/api/batch/gen.github.a9901fd7-d88c-496b-a2ba-06dc0b83c515?csBaseUrl=https%3A%2F%2Fokta.acrolinx.cloud&X-Acrolinx-Client-Locale=en)

File: packages/@okta/vuepress-site/docs/reference/token-hook/index.md [hunk 3]
```diff
@@ -50,6 +50,14 @@ Provides information on the properties of the access token that Okta has generat
 | lifetime | Lifetime of the token.             | [lifetime](#lifetime) object |
 | scopes   | The scopes contained in the token. | [scopes](#scopes) object     |
 
+#### data.refresh_token
+
+Provides the refresh token ID that you can store on the first access token request. When you make the refresh token grant, the ID matches what was stored. This allows you to connect the two requests.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/token-hook/index.md:55]: @venkatviswanathan-okta and @anweshdatla-okta - I updated the sentence

**Comment by acrolinxdevdocs-okta** [comment]
## Acrolinx score **DRAFT 06.01.22**
 
A minimum Acrolinx Score of **40** is required. The total score is an average of the subscores.

Select **Total score** to review the Acrolinx scorecard for your article. Try to increase your individual scores, ex. Correctness. Your article will be clearer and more consistent with Microsoft standards.
 
| Article | **Total score**<br />**Required:70** | Word and phrases<br />(Brand, terms)<br />Preferred: 40 | Correctness<br />(Spelling, grammar)<br />Preferred: 40 | Clarity<br />(Readability)<br />Preferred: 10 | Inclusive language<br />(+ accesibility)<br />Preferred: 10 |
|---------|:------------------------------------:|:--------------------------------------------------------:|:--------------------------------------------------------:|:---------------------------------------------:|:------------------------------------------------------------:|
| [:white_check_mark:](# "Successfully processed") [packages/@okta/vuepress-site/docs/reference/token-hook/index.md](https://github.com/okta/okta-developer-docs/blob/d7ea52b07cfd57cfe77580ac122a025f3afc2027/packages%2F%40okta%2Fvuepress-site%2Fdocs%2Freference%2Ftoken-hook%2Findex.md) | [68](https://okta.acrolinx.cloud/api/v1/checking/scorecards/45b2db3e-1764-4317-9ab0-256e6af3eaac) | 80  | 57  | 34  | 96  |

  - The results from this response were procced by the [hugo-targets1.solution.acrolinx.sh](https://hugo-targets1.solution.acrolinx.sh) Acrolinx Cloud Host.
  - Scorecard links by default are only good for 7 days. [Link for this commit's Content Analysis Dashboard](https://okta.acrolinx.cloud/api/batch/gen.github.c12c75a5-e18f-4f66-a7ca-7f3f01703167?csBaseUrl=https%3A%2F%2Fokta.acrolinx.cloud&X-Acrolinx-Client-Locale=en)

File: packages/@okta/vuepress-site/docs/reference/token-hook/index.md [hunk 4]
```diff
@@ -64,6 +64,14 @@ Consists of name-value pairs for each included claim. For descriptions of the cl
 
 The set of scopes that have been granted. For descriptions of the scopes that can be included, see Okta's [OpenID Connect and OAuth 2.0 API reference](/docs/reference/api/oidc/#tokens-and-claims).
 
+#### data.refresh_token
+
+Provides the refresh token ID that you can store on the first access token request. When you make the refresh token grant, the ID matches what was stored. This allows you to connect the two requests.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/token-hook/index.md:69]: @venkatviswanathan-okta and @anweshdatla-okta please review this. Thanks.

File: packages/@okta/vuepress-site/docs/reference/token-hook/index.md [hunk 5]
```diff
@@ -64,6 +64,14 @@ Consists of name-value pairs for each included claim. For descriptions of the cl
...
+
+| Property   | Description                              | Data Type |
+|------------|------------------------------------------|-----------|
+| jti        | Refresh token ID                         | Number    |
```
Comment (anweshdatla-okta) [inline packages/@okta/vuepress-site/docs/reference/token-hook/index.md:73]:
```suggestion
| jti        | Refresh token ID                         | String    |
```

**Review by venkatviswanathan-okta — APPROVED** [review]

**Comment by acrolinxdevdocs-okta** [comment]
## Acrolinx score **DRAFT 06.01.22**
 
A minimum Acrolinx Score of **40** is required. The total score is an average of the subscores.

Select **Total score** to review the Acrolinx scorecard for your article. Try to increase your individual scores, ex. Correctness. Your article will be clearer and more consistent with Microsoft standards.
 
| Article | **Total score**<br />**Required:70** | Word and phrases<br />(Brand, terms)<br />Preferred: 40 | Correctness<br />(Spelling, grammar)<br />Preferred: 40 | Clarity<br />(Readability)<br />Preferred: 10 | Inclusive language<br />(+ accesibility)<br />Preferred: 10 |
|---------|:------------------------------------:|:--------------------------------------------------------:|:--------------------------------------------------------:|:---------------------------------------------:|:------------------------------------------------------------:|
| [:white_check_mark:](# "Successfully processed") [packages/@okta/vuepress-site/docs/reference/token-hook/index.md](https://github.com/okta/okta-developer-docs/blob/1dc29ee7ca8a92cadfdd4c0c82dc1fc59e2c538a/packages%2F%40okta%2Fvuepress-site%2Fdocs%2Freference%2Ftoken-hook%2Findex.md) | [68](https://okta.acrolinx.cloud/api/v1/checking/scorecards/5a60ad14-3059-4ff2-b55e-52cc4667892a) | 80  | 57  | 34  | 96  |

  - The results from this response were procced by the [hugo-targets1.solution.acrolinx.sh](https://hugo-targets1.solution.acrolinx.sh) Acrolinx Cloud Host.
  - Scorecard links by default are only good for 7 days. [Link for this commit's Content Analysis Dashboard](https://okta.acrolinx.cloud/api/batch/gen.github.9401e274-729a-492a-a612-8ab1126ece30?csBaseUrl=https%3A%2F%2Fokta.acrolinx.cloud&X-Acrolinx-Client-Locale=en)

**Review by anweshdatla-okta — APPROVED** [review]

**Pull request merged**